Ghidra/Features/Decompiler/src/decompile/cpp/sleigh_arch.cc

```diff
@@ -174,17 +174,15 @@ bool SleighArchitecture::isTranslateReused(void)
 Translate *SleighArchitecture::buildTranslator(DocumentStorage &store)
 
 {				// Build a sleigh translator
-  map<int4,Sleigh *>::const_iterator iter;
-  Sleigh *sleigh;
+  map<int4,Sleigh>::iterator iter;
+
   iter = translators.find(languageindex);
   if (iter != translators.end()) {
-    sleigh = (*iter).second;
-    sleigh->reset(loader,context);
-    return sleigh;
+    iter->second.reset(loader, context);
+    return &iter->second;
   }
-  sleigh = new Sleigh(loader,context);
-  translators[languageindex] = sleigh;
-  return sleigh;
+  translators[languageindex] = Sleigh(loader,context);
```

Comment:
Please use the emplace method to avoid the need for a copy constructor and a default constructor for the Sleigh class.

```
  pair<map<int4,Sleigh>::iterator,bool> res;
  res = translators.emplace(piecewise_construct,forward_as_tuple(languageindex),forward_as_tuple(loader,context));
  return &(*res.first).second;

```

Reply:
Done in https://github.com/NationalSecurityAgency/ghidra/pull/6890/commits/f56473e6047e3086d69be250f372fd7657efe5bd.